arangod/Agency/Node.h

```diff
@@ -75,6 +75,60 @@ typedef std::chrono::steady_clock::time_point SteadyTimePoint;
 
 class Store;
 
+class SmallBuffer {
+  uint8_t* _start;
+  size_t _size;
+ public:
+  SmallBuffer() : _start(nullptr), _size(0) {}
+  SmallBuffer(size_t size) : _start(new uint8_t[size]), _size(size) {}
+  SmallBuffer(SmallBuffer const& other) {
+    if (other.empty()) {
+      _start = nullptr;
+      _size = 0;
+    } else {
+      _start = new uint8_t[other._size];
+      _size = other._size;
+      memcpy(_start, other._start, other._size);
+    }
+  }
+  SmallBuffer(SmallBuffer&& other) {
+    _start = other._start;
+    other._start = nullptr;
+    _size = other._size;
+    other._size = 0;
+  }
+  SmallBuffer& operator=(SmallBuffer const& other) {
```

Comment:
This would be broken if `this == &other`.
Should we add the usual `if (this != &other) { ... }` check here?

Reply:
Yes, this is a good idea, will do.

Reply:
Done.